File: pytools/examples/php/server/protocol/ReqChatSend.php
<?php
namespace protocol;

use network\Packet;


class ReqChatSend
{
	private $channel = 0;
	private $destUid = 0;
	private $content = '';


	public function __construct(&$packet)
	{
		$this->decode($packet);
	}

	public function decode(&$packet)
	{
		$this->channel = $packet->readU8();
		$this->destUid = $packet->readU32();
		$this->content = $packet->readString();
	}


	public function getChannel()
	{
		return $this->channel;
	}

	public function getDestUid()
	{
		return $this->destUid;
	}

	public function getContent()
	{
		return $this->content;
	}

}

File: pytools/examples/java/server/protocol/ReqChatSend.java
package protocol;


import network.Packet;
import network.PacketUtil;

import java.util.List;
import java.util.ArrayList;


public class ReqChatSend
{
	private byte channel;
	private int dest_uid;
	private String content;


	public ReqChatSend(Packet packet)
	{
		channel = packet.readByte();
		dest_uid = packet.readInt();
		content = packet.readString();
	}



	public int getChannel()
	{
		return PacketUtil.readUByte(this.channel);
	}

	public long getDestUid()
	{
		return PacketUtil.readUInt(this.dest_uid);
	}

	public String getContent()
	{
		return this.content;
	}

}
